core/src/main/java/org/mule/config/pool/DefaultThreadPoolFactory.java

```diff
@@ -81,8 +84,9 @@ private void configureThreadPoolExecutor(String name, ThreadingProfile tp, Threa
     @Override
     public ScheduledThreadPoolExecutor createScheduledPool(String name, ThreadingProfile tp)
     {
-        ScheduledThreadPoolExecutor pool = internalCreateScheduledPool(tp);
-        configureThreadPoolExecutor(name, tp, pool);
+        ThreadingProfile threadingProfileCopy = tp instanceof DirectThreadingProfile ? tp : new ImmutableThreadingProfile(tp);
```

Comment:
Yes, but not in scope of this PR. And given this PR has config changes that we need to make today..